packages/core/pvm-interpreter-ananas/index.ts

```diff
@@ -0,0 +1,119 @@
+import { tryAsU32, tryAsU64, type U64 } from "@typeberry/numbers";
+import type { IHostCallMemory, IHostCallRegisters } from "@typeberry/pvm-host-calls";
+import { type Gas, gasCounter, Status, tryAsBigGas, tryAsGas, tryAsMemoryIndex } from "@typeberry/pvm-interpreter";
+import { type OutOfBounds, PageFault } from "@typeberry/pvm-interpreter/memory/errors.js";
+import { OK, Result } from "@typeberry/utils";
+import {
+  disassemble,
+  getExitArg,
+  getGasLeft,
+  getMemory,
+  getRegister,
+  getStatus,
+  HasMetadata,
+  InputKind,
+  nextStep,
+  resetJAM,
+  setGasLeft,
+  setMemory,
+  setRegister,
+} from "anan-as";
+
+export class AnanasInterpreter implements IHostCallRegisters, IHostCallMemory {
+  private pc = 0;
+  private initialGas = gasCounter(tryAsGas(0n));
+
+  resetJam(program: Uint8Array, args: Uint8Array, pc: number, gas: Gas) {
+    const programArr = lowerBytes(program);
+    const argsArr = lowerBytes(args);
+    this.pc = pc;
+    this.initialGas = gasCounter(gas);
+    resetJAM(programArr, this.pc, BigInt(gas), argsArr, true);
+  }
+
+  printProgram(program: Uint8Array) {
+    return disassemble(lowerBytes(program), InputKind.SPI, HasMetadata.Yes);
+  }
+
+  runProgram() {
+    while (nextStep()) {}
+  }
+
+  getStatus() {
+    const status = getStatus();
+    if (status < 0) {
+      return Status.OK;
+    }
+    return status;
+  }
+
+  getPC() {
+    return this.pc;
+  }
+
+  getGas() {
+    return tryAsGas(getGasLeft());
+  }
+
+  setGas(gas: Gas) {
+    setGasLeft(gas as bigint);
+  }
+
+  subGas(gas: Gas) {
+    const newGas = gasCounter(this.getGas());
+    const result = newGas.sub(gas);
+    setGasLeft(newGas.get() as bigint);
+    return result;
+  }
+
+  getExitParam() {
+    const arg = getExitArg();
```

Comment:
Why is the conditional logic needed here? exit param should be irrelevant unless the machine stopped at a host call or made a page fault.